phoenix-core/src/main/java/org/apache/phoenix/compile/FromCompiler.java

```diff
@@ -1121,46 +1124,55 @@ public PSchema resolveSchema(String schemaName) throws SQLException {
+        protected final List<TableRef> localIndexDataTables = new ArrayList<>();;
         private final List<TableRef> theTableRefs;
         private final Map<ColumnRef, Integer> columnRefMap;
         private ProjectedTableColumnResolver(PTable projectedTable, PhoenixConnection conn, Map<String, UDFParseNode> udfParseNodes) throws SQLException {
+
             super(conn, 0, udfParseNodes, null);
             Preconditions.checkArgument(projectedTable.getType() == PTableType.PROJECTED);
             this.isLocalIndex = projectedTable.getIndexType() == IndexType.LOCAL;
             this.columnRefMap = new HashMap<ColumnRef, Integer>();
             long ts = Long.MAX_VALUE;
             for (int i = projectedTable.getBucketNum() == null ? 0 : 1; i < projectedTable.getColumns().size(); i++) {
                 PColumn column = projectedTable.getColumns().get(i);
-                ColumnRef colRef = ((ProjectedColumn) column).getSourceColumnRef();
-                TableRef tableRef = colRef.getTableRef();
-                if (!tables.contains(tableRef)) {
-                    String alias = tableRef.getTableAlias();
-                    if (alias != null) {
-                        this.tableMap.put(alias, tableRef);
-                    }
-                    String name = tableRef.getTable().getName().getString();
-                    if (alias == null || !alias.equals(name)) {
-                        tableMap.put(name, tableRef);
-                    }
-                    tables.add(tableRef);
-                    if (tableRef.getLowerBoundTimeStamp() < ts) {
-                        ts = tableRef.getLowerBoundTimeStamp();
-                    }
+                ColumnRef sourceColRef = ((ProjectedColumn) column).getSourceColumnRef();
+                TableRef tableRef = sourceColRef.getTableRef();
+
+                if (sourceColRef instanceof LocalIndexDataColumnRef && !localIndexDataTables.contains(tableRef)) {
+                    rememberLocalIndexDataTable(tableRef, tableRef.getTableAlias());
+                } else
+                    if (!tables.contains(tableRef)) {
+                    rememberTable(tableRef, tableRef.getTableAlias());
+                }
+                if (tableRef.getLowerBoundTimeStamp() < ts) {
+                    ts = tableRef.getLowerBoundTimeStamp();
                 }
-                this.columnRefMap.put(new ColumnRef(tableRef, colRef.getColumnPosition()), column.getPosition());
+                this.columnRefMap.put(new ColumnRef(tableRef, sourceColRef.getColumnPosition()), column.getPosition());
             }
             this.theTableRefs = ImmutableList.of(new TableRef(ParseNodeFactory.createTempAlias(), projectedTable, ts, false));
-            
         }
-        
+
+        protected void rememberLocalIndexDataTable(TableRef tableRef, String alias) {
+            String name = tableRef.getTable().getName().getString();
```

Comment:
nit: similar to `rememberTable`, this can also be put inside else block

Reply:
Done.